conans/client/cache/cache.py

```diff
@@ -100,7 +100,13 @@ def package_layout(self, ref, short_paths=None):
         if edited_ref:
             base_path = edited_ref["path"]
             layout_file = edited_ref["layout"]
-            return PackageEditableLayout(base_path, layout_file, ref)
+            if os.path.isfile(base_path):
+                conanfile_name = os.path.basename(base_path)
+                base_path = os.path.dirname(base_path)
+                return PackageEditableLayout(base_path, layout_file, ref, conanfile_name)
+            else:
+                # FIXME: Remove in Conan 2.0, introduced for <= 1.25 backward compatibility
+                return PackageEditableLayout(base_path, layout_file, ref)
```

Comment:
To support existing `editable_packages.json` files. May I forget about this, may I write a migration?